client/rest-high-level/src/test/java/org/elasticsearch/client/RestHighLevelClientTests.java

```diff
@@ -751,8 +751,11 @@ public void testApiNamingConventions() throws Exception {
                     assertThat("the return type for method [" + method + "] is incorrect",
                         method.getReturnType().getSimpleName(), equalTo("boolean"));
                 } else {
-                    assertThat("the return type for method [" + method + "] is incorrect",
-                        method.getReturnType().getSimpleName(), endsWith("Response"));
+                    // It's acceptable for 404s to be represented as empty Optionals 
+                    if (!method.getReturnType().isAssignableFrom(Optional.class)) {
```

Comment:
can we ensure that the thing *in* the optional endsWith("Response"), as well?

Reply:
I did have a stab at that and I'm not sure I can get at it via reflection. The best I could get was that the Optional class used in the response has a generic type of `T`. I suspect type erasure might prevent me from seeing that level of detail at runtime.

Reply:
ok, i was unsure myself :)